Gems/Atom/RHI/Vulkan/Code/Source/RHI/Image.cpp

```diff
@@ -188,6 +618,197 @@ namespace AZ
+            createInfo.sType = VK_STRUCTURE_TYPE_IMAGE_CREATE_INFO;
+            createInfo.pNext = nullptr;
+            createInfo.format = ConvertFormat(descriptor.m_format);
+            createInfo.flags = GetImageCreateFlags();
+            // Add flags for sparse binding and sparse memory residency
+            createInfo.flags |= VK_IMAGE_CREATE_SPARSE_BINDING_BIT | VK_IMAGE_CREATE_SPARSE_RESIDENCY_BIT; 
+            createInfo.imageType = ConvertToImageType(descriptor.m_dimension);
+            createInfo.tiling = VK_IMAGE_TILING_OPTIMAL;
+            createInfo.usage = GetImageUsageFlags();
+
+            // Get general image properties
+            VkImageFormatProperties formatProps{};
+            AssertSuccess(device.GetContext().GetPhysicalDeviceImageFormatProperties(
+                physicalDevice.GetNativePhysicalDevice(), createInfo.format, createInfo.imageType, createInfo.tiling, createInfo.usage,
+                createInfo.flags, &formatProps));
+
+            VkSampleCountFlagBits sampleCountFlagBits = static_cast<VkSampleCountFlagBits>(RHI::FilterBits(static_cast<VkSampleCountFlags>(ConvertSampleCount(descriptor.m_multisampleState.m_samples)), formatProps.sampleCounts));
+            createInfo.samples = (static_cast<uint32_t>(sampleCountFlagBits) > 0) ? sampleCountFlagBits : VK_SAMPLE_COUNT_1_BIT;
+            
+            VkExtent3D extent = ConvertToExtent3D(descriptor.m_size);
+            extent.width = AZStd::min<uint32_t>(extent.width, formatProps.maxExtent.width);
+            extent.height = AZStd::min<uint32_t>(extent.height, formatProps.maxExtent.height);
+            extent.depth = AZStd::min<uint32_t>(extent.depth, formatProps.maxExtent.depth);
+            createInfo.extent = extent;
+            createInfo.mipLevels = AZStd::min<uint32_t>(descriptor.m_mipLevels, formatProps.maxMipLevels);
+            createInfo.arrayLayers = AZStd::min<uint32_t>(descriptor.m_arraySize, formatProps.maxArrayLayers);
+
+            // We will always handles image's ownership
+            createInfo.sharingMode = VK_SHARING_MODE_EXCLUSIVE;
+            createInfo.queueFamilyIndexCount = 1;
+            createInfo.pQueueFamilyIndices = nullptr;
+
+            createInfo.initialLayout = VK_IMAGE_LAYOUT_UNDEFINED;
+
+            const VkResult result = device.GetContext().CreateImage(device.GetNativeDevice(), &createInfo, nullptr, &m_vkImage);
+
+            if (result == VkResult::VK_SUCCESS)
+            {
+                // If the sparse image's alignment is not standard block size, release the image and return fail
+                // We only support images with standard block size due to memory allocator restriction
+                VkMemoryRequirements memoryRequirements;
+                device.GetContext().GetImageMemoryRequirements(device.GetNativeDevice(), m_vkImage, &memoryRequirements);
+                if (memoryRequirements.alignment != SparseImageInfo::StandardBlockSize)
```

Comment:
Shouldnt this be checked before we call createimage?

Reply:
Nope. The GetImageMemoryRequirements function only takes a VKImage object as parameter. This is because even the device support standard block size, it doesn't mean all the images would be able to use the standard block size.